work_queue/src/bindings/python3/PyTask/PyTask.py

```diff
@@ -0,0 +1,125 @@
+		self._args_file = os.path.join(self._tmpdir, 'args_{}.p'.format(self._id))
+		self._out_file = os.path.join(self._tmpdir, 'out_{}.p'.format(self._id))
+		self._wrapper = os.path.join(self._tmpdir, 'pytask_wrapper.py'.format(self._id))
+		#self._package_file = os.path.join(self._tmpdir, 'dep.yml')
+		self._tar_file  = 'venv.tar.gz'
+		self._pp_run = shutil.which('python_package_run')		
+
+		self._command = self.python_function_command(func, *args)
+		self.create_wrapper()
+
+		super(PyTask, self).__init__(self._command)
+		
+		self.specify_IO_files()
+		
+
+		PyTask.p_tasks.append(self)
+		
+	
+	def python_function_command(self, func, *args):
+			
+	
+		tb = traceback.extract_stack()
+		tb_list = traceback.format_list(tb)
+		caller = tb_list[len(tb_list)-3].split()[1]
+		caller = caller.replace('"', '')
+		caller = caller.replace(',', '')
+
+
+	#	if not os.path.exists(tar_file):
+	#		os.system('python_package_analyze ' + caller + ' ' + package_file)
+
+	#	os.system('python_package_create x.yml ' + self._tar_file)
+			
+		
+		with open(self._func_file, 'wb') as wf:
+			dill.dump(func, wf)
+		
+		with open(self._args_file, 'wb') as wf:
+			dill.dump([*args], wf)
+		
+		#command = "python {wrapper} {function} {args} {out}".format(
+		command = "./{pprun} -e {tar} python {wrapper} {function} {args} {out}".format(
+			pprun=os.path.basename(self._pp_run),
+			tar=os.path.basename(self._tar_file),
+			wrapper=os.path.basename(self._wrapper),
+			function=os.path.basename(self._func_file),
+			args=os.path.basename(self._args_file),
+			out=os.path.basename(self._out_file))
+
+
+		return command
+
+	def specify_IO_files(self):
+
+		self.specify_input_file(self._wrapper, cache=True)
+		self.specify_input_file(self._pp_run, cache=True)
+		self.specify_input_file(self._tar_file, cache=False)
+		self.specify_input_file(self._func_file, cache=False)
+		self.specify_input_file(self._args_file, cache=False)
+		self.specify_output_file(self._out_file, cache=False)
+
+	def create_wrapper(self):
+		
+		with open(self._wrapper, 'w') as f:
+			f.write(textwrap.dedent('''\
+				import sys
+				import dill
+				(fn, args, out) = sys.argv[1], sys.argv[2], sys.argv[3]
+				with open (fn , 'rb') as f:
+					exec_function = dill.load(f)
+				with open(args, 'rb') as f:
+					exec_args = dill.load(f)
+				try:
+					exec_out = exec_function(*exec_args)
+
+				except Exception as e:
+					exec_out = e
+
+				with open(out, 'wb') as f:
+					dill.dump(exec_out, f)
+
+				print(exec_out)'''))
+			 
+				
+	@staticmethod
+	def python_result(task):		
+		if task.result == wq.WORK_QUEUE_RESULT_SUCCESS:
+			try:
+				with open(os.path.join(task._tmpdir, 'out_{}.p'.format(task._id)), 'rb') as f:
+					func_result = dill.load(f)
+			except Exception as e:
+				func_result = e
+		else:
+			print('no reuslt for task {}, error code {} {}'.format(task.tag, task.result, task.result_str))
+			func_result = 'no result'
```

Comment:
For this one, let's create a class:
```python
PyTaskNoResult(Exception):
   pass
```

and return that instead. We can even add task.result and result_str as fields of the class.